### DISCUSSION TITLE
Inline Property Value Maths

### DISCUSSION THREAD

**Discussion by NworbLegin** [mapeditor/tiled#3629] [Ideas]
When a property is a numeric value it would be great to be able to perform basic maths operations as part of the data entry.

For example:
An object has a y position of 1654 and I want to move the item up by 138 pixels.
I could calculate the value of  1654 - 138 = 1516 and enter 1516 into the edit box, but it would be easier to select on the 1654 value and at the end of the line type "- 138" and let the value be calculated automatically.
Operators for basic addition, subtraction, multiplication, and division could be added to achieve this.

What does anyone else think to this?

**Comment by bjorn** [comment]
I really like the suggestion, but unfortunately this may not be something that's easy to support using the standard number inputs that Qt provides, since it's not possible to type anything in them that would not parse as a valid number. But maybe there is a solution available.

**Comment by NworbLegin** [comment]
For now I'm executing this script line in the console after selecting items:

`map = tiled.activeAsset; for (index in map.selectedObjects) {map.selectedObjects[index].height -= 66;}`

... which works okay - and is probably a little more flexible as it will operate on multiple objects.

**Reply by bjorn** [reply]
That works very well indeed. You might consider wrapping the for loop in a `map.macro(() => { for ... })`, such that all the height changes will be part of a single undo command.
